rs/protobuf/def/state/canister_state_bits/v1/canister_state_bits.proto

```diff
@@ -312,12 +312,28 @@ message WasmChunkStoreMetadata {
   uint64 size = 2;
 }
 
+// TODO(EXC-1670): Migrate to pb_canister_state_bits::LogVisibilityV2.
+// The current enum only supports i32 values, which limits the
+// storage of allowed_viewers principals. LogVisibilityV2 will
+// support both enum values and a list of principals.
 enum LogVisibility {
   LOG_VISIBILITY_UNSPECIFIED = 0;
   LOG_VISIBILITY_CONTROLLERS = 1;
   LOG_VISIBILITY_PUBLIC = 2;
 }
 
+message LogVisibilityAllowedViewers {
+  repeated types.v1.PrincipalId principals = 1;
+}
+
+message LogVisibilityV2 {
+  oneof log_visibility_v2 {
+    int32 controllers = 1;
+    int32 public = 2;
+    LogVisibilityAllowedViewers allowed_viewers = 3;
+  }
+}
```

Comment:
Personal opinion, feel free to ignore: This is a somewhat unusual choice: what is the value of `controllers` or `public` supposed to mean? Why not simply have a discriminator (an integer describing which variant this is; as Rust structs do) plus the union of optional fields required by all variants?

Actually, it may be simpler (and require fewer changes) to change the types of `controllers` and `public` from `int32` to `google.protobuf.Empty` (or some custom empty types you define here).

Reply:
I would prefer to keep it as it is now, because it's semantically the closest we can get to Rust version, and also because we use the same approach for other enum cases.